[Ryan Aidan] iP

aidanaden opened the pull request:
### DukePro2
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro2 frees your mind of having to remember things you need to do. It's

* text-based
* easy to learn
* ~FAST~ _SUPER_ FAST to use

All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 👍

And it is **FREE**

Features:
* [x]  Managing tasks
* [x]  Managing deadlines
* [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
added greet, echo, exit with base styling

Commit:
added lvl2 with Add, List

Commit:
Added level-3, Mark as done

Commit:
Added level-4, Todos/Events/Deadlines + enums

Commit:
added level-5, exception handling

Commit:
added level-6, delete

Commit:
Add automated text-ui-text lvl 6

Commit:
Update exceptions with javadocs

Commit:
Add Level-7 Save to hard drive

Commit:
Add Level-8 store timings with LocalDateTime and use Event/Todo/Type subclasses

aidanaden committed:
Merge pull request #1 from aidanaden/branch-Level-7

Add Level-7 Save to hard drive

Commit:
Fix merge conflicts for master and Level-8 branch

Commit:
Fix merge conflicts for master and Level-8 branch

Commit:
Add more OOP including Storage, Tasklist, Ui, Command + Type enums

aidanaden committed:
Merge pull request #3 from aidanaden/A-MoreOOP

Add more OOP including Storage, Tasklist, Ui, Command + Type enums

Commit:
Add duke package

Commit:
Fix merge conflicts from packaging duke

aidanaden committed:
Restart with gradle from A-MoreOOP

aidanaden committed:
Fix merge conflict from gradle initialisation

aidanaden committed:
Deadline: removed typos

aidanaden committed:
Duke: removed merge conflict typos

aidanaden committed:
Add javadoc to all java classes & functions except Duke + tests

aidanaden committed:
Add JUnit test for Deadline, Event, Parser, Todo

aidanaden committed:
Update build gradle with jarShadow config

aidanaden committed:
Add find feature, todo: add testing + error handling

aidanaden committed:
Merge branch 'branch-A-JavaDoc' into master

aidanaden committed:
Merge branch 'branch-Level-9' into master

atmh commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,17 @@
+/**
+ * Command types enum
+ */
+public enum Command {
+    LIST,
+    MARK,
+    UNMARK,
+    DEADLINE,
+    EVENT,
+    TODO,
+    DELETE,
+    FIND,
+    BYE,
+    ERROR
+}
```
I like that you have included enum 👍.

atmh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,46 @@
+        String[] inputSplit = input.split(" ");
+
+        if (inputSplit.length > 0) {
+            String commandString = inputSplit[0];
+
+            if (commandString.equals("bye")) {
+                return Command.BYE;
+            } else if (commandString.equals("list")) {
+                return Command.LIST;
+            } else if (commandString.equals("mark")) {
+                return Command.MARK;
+            } else if (commandString.equals("unmark")) {
+                return Command.UNMARK;
+            } else if (commandString.equals("deadline")) {
+                return Command.DEADLINE;
+            } else if (commandString.equals("event")) {
+                return Command.EVENT;
+            } else if (commandString.equals("todo")) {
+                return Command.TODO;
+            } else if (commandString.equals("delete")) {
+                return Command.DELETE;
+            } else if (commandString.equals("find")) {
+                return Command.FIND;
+            } else {
+                return Command.ERROR;
+            }
```
Perhaps a switch statement would be more succinct. E.g. 
```java
switch (commandString) {
case "bye":
    ...
    break;
case "list":
    ...
    break;
default:
    ...
}
```

aidanaden replied:
but switch statements look more complicated to me :(

atmh commented on src/main/java/duke/DeleteEmptyException.java:
```diff
@@ -0,0 +1,11 @@
+ * Exception when user attempts to delete a task
+ * without providing the index of the task
+ */
+public class DeleteEmptyException extends DukeException {
```
I like that you have created classes for specific exceptions

atmh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,154 @@
+            tasks = new TaskList();
+        }
+    }
+
+    public String[] getEventTimings(String text) {
```
Just a little more Javadocs to go 👍

atmh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,147 @@
+package duke;
```
Although it's not required, I feel that dividing the classes into multiple packages would be cleaner. 
E.g. `duke.storage`, `duke.task`, `duke.ui`, `duke.command`

atmh commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,92 @@
+    /**
+     * Completion status of task
+     */
+    protected boolean isDone;
```
Overall, the code follows the naming guidelines very closely 👍.

atmh reviewed (commented):
Looks good 😄.
The code abides to the code quality guidelines 👍, just some minor nits to clean up.

aidanaden committed:
Merge branch 'A-Jar' into master

aidanaden committed:
Add checkstyle 8.1

tanchinkiat99 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,22 @@
+ */
+public class DukeException extends Exception {
+    public DukeException(String err) {
+        super("\n############################################################\n ☹ OOPS!!! "
```
Will it be better to have a more comprehensive error message?

tanchinkiat99 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,114 @@
+     * @return Start datetime as a string using the default format
+     */
+    public String getStartTimeString() {
+        return getStartTimeString("dd/MM/yyyy HHmm");
```
Will it be better to combine this with the above function?

tanchinkiat99 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,46 @@
+        if (inputSplit.length > 0) {
+            String commandString = inputSplit[0];
+
+            if (commandString.equals("bye")) {
```
Is it more suitable to use enumeration and switch function here?

tanchinkiat99 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,147 @@
+                String taskName = taskLineSplit[2];
+                Task storedTask;
+
+                if (taskLineSplit[0].equals("T")) {
```
I like how you used comments to show the functionality of each statement. 😄

tanchinkiat99 reviewed (commented):
Looks good. Just a few possible improvements here and there to help make your code more organised.

aidanaden committed:
Add Level-10 GUI

aidanaden committed:
Undo gui testing

aidanaden committed:
Fix Level-10 merge conflicts

aidanaden committed:
Add assertions

aidanaden committed:
Merge pull request #4 from aidanaden/branch-A-Assertions

Add assertions

aidanaden committed:
Refactor heavily

aidanaden committed:
Merge pull request #5 from aidanaden/branch-A-CodeQuality

Refactor heavily

aidanaden committed:
Added tagging with storage

Task class: Stores task data such as name/description, timing data,
mark/unmark status.

Tasks can be categorised into different priority levels (e.g. urgent,
important, optional, etc).

There are no possible way to categorise tasks.

Adding a list of tags to the Task class allows tasks to be categorised
into various different categories. For example, an event can be tagged
as "optional" and "expiring" and thus fit in both categories.

Let's add an arraylist of strings to represent the tags associated with
the Task class.

Using an arraylist of strings to store the tags is a convenient way
of storing and displaying tags of tasks. May create a dedicated Tag
class in the future instead.

aidanaden committed:
Merge pull request #6 from aidanaden/branch-C-Tagging

Added tagging with storage

aidanaden committed:
Enforce coding standards across all files

aidanaden committed:
Add exit with delay to bye command

aidanaden committed:
Update readme with program features and instructions

aidanaden committed:
Update readme with release URL

aidanaden committed:
Add UI png

aidanaden committed:
Fix UI png file extension all caps

aidanaden committed:
Update docs readme file

aidanaden committed:
Refactor code according to code quality feedback from nus-se-bot
